frontend/src/pages/FindChain.tsx

```diff
@@ -116,6 +124,71 @@ const FindChain = () => {
     }
   };
 
+  const handleMapClick = (event: MapEvent) => {
+    const topMostFeature = event?.features?.[0];
+    const {
+      layer: { id: layerId },
+    } = topMostFeature;
+
+    if (layerId === 'chains') {
+      const selectedChainIndex = topMostFeature.properties.chainIndex;
+
+      setSelectedChain(filteredChains[selectedChainIndex]);
+      setShowPopup(true);
+    } else if (layerId === 'cluster' || layerId === 'cluster-count') {
+      const clusterId = topMostFeature.properties.cluster_id;
+
+      const mapboxSource = mapRef!.current!.getMap().getSource('chains');
+
+      mapboxSource.getClusterExpansionZoom(
+        clusterId,
+        (err: any, zoom: number) => {
+          if (err) {
+            return;
+          }
+
+          const {
+            geometry: { coordinates },
+          } = topMostFeature;
+
+          setViewport({
+            ...viewport,
+            longitude: coordinates[0],
+            latitude: coordinates[1],
+            zoom,
+            transitionDuration: 500,
+          });
+        }
+      );
+    }
+  };
+
+  const geoJSONFilteredChains: GeoJSONTypes.FeatureCollection<GeoJSONTypes.Geometry> =
+    {
+      type: 'FeatureCollection',
+      features: filteredChains
+        .filter(({ published }) => published)
+        .map((filteredChain, filteredChainIndex) => {
+        const {
+          longitude,
+          latitude,
+          categories: { gender },
+        } = filteredChain;
+
+        return {
+          type: 'Feature',
+          geometry: {
+            type: 'Point',
+            coordinates: [longitude, latitude],
+          },
+          properties: {
+            chainIndex: filteredChainIndex,
+            gender: gender[0], // GeoJSON doesn't support nested array, see https://github.com/mapbox/mapbox-gl-js/issues/2434
```

Comment:
This logic is different from previous implementation, which is
`chain.categories.gender.includes("women")`

I have rough idea how to implement the previous logic, but I think we can discuss further whether we want to retain previous implementation? Especially if a chain may have multiple gender categories